Libplanet/Store/LiteDBStore.cs

```diff
@@ -510,136 +504,39 @@ public long GetTxNonce(string @namespace, Address address)
-                _db.FileStorage
-                    .Find($"{NonceIdPrefix}{sourceNamespace}")
-                    .ToList();
-
-            if (!files.Any() && addressesToStrip.Any())
-            {
-                throw new NamespaceNotFoundException(
-                    sourceNamespace,
-                    "The source namespace to be forked does not exist.");
-            }
+            var fileId = $"{NonceIdPrefix}{@namespace}/{signer.ToHex()}";
+            long nextNonce = GetTxNonce(@namespace, signer) + delta;
 
-            foreach (LiteFileInfo srcFile in files)
+            if (!_db.FileStorage.Exists(fileId))
             {
-                string destId =
-                    $"{NonceIdPrefix}{destinationNamespace}/{srcFile.Filename}";
                 _db.FileStorage.Upload(
-                    destId,
-                    srcFile.Filename,
+                    fileId,
+                    signer.ToHex(),
                     new MemoryStream());
+            }
 
-                LiteFileInfo destFile = _db.FileStorage.FindById(destId);
-                using (LiteFileStream srcStream = srcFile.OpenRead())
-                using (LiteFileStream destStream = destFile.OpenWrite())
-                {
-                    while (srcStream.Position < srcStream.Length)
-                    {
-                        var hashBytes = new byte[HashDigest<SHA256>.Size];
-                        var indexBytes = new byte[sizeof(long)];
-                        var nonceBytes = new byte[sizeof(long)];
-
-                        srcStream.Read(hashBytes, 0, hashBytes.Length);
-                        srcStream.Read(indexBytes, 0, indexBytes.Length);
-                        srcStream.Read(nonceBytes, 0, nonceBytes.Length);
-
-                        long currentIndex =
-                            BitConverter.ToInt64(indexBytes, 0);
-
-                        if (currentIndex <= branchPointIndex)
-                        {
-                            destStream.Write(hashBytes, 0, hashBytes.Length);
-                            destStream.Write(indexBytes, 0, indexBytes.Length);
-                            destStream.Write(nonceBytes, 0, nonceBytes.Length);
-                        }
-                        else
-                        {
-                            break;
-                        }
-                    }
-                }
+            LiteFileInfo file = _db.FileStorage.FindById(fileId);
+            using (var temp = new MemoryStream())
+            {
+                file.CopyTo(temp);
+                temp.Seek(0, SeekOrigin.Begin);
+                byte[] prev = temp.ToArray();
 
-                if (destFile.Length == 0)
+                using (LiteFileStream stream = file.OpenWrite())
                 {
-                    _db.FileStorage.Delete(destId);
+                    byte[] nonceBytes = BitConverter.GetBytes(nextNonce);
+                    stream.Write(nonceBytes, 0, nonceBytes.Length);
```

Comment:
It would be also better if we use `LiteDBCollection` instead of file storage because the nonce became to simple `long`. (even later)

Reply:
bedbe8a651b328c281b051b419a64fa281dd8959